src/settings/components/utils.ts

```diff
@@ -496,13 +496,39 @@ export function defaultAsFirstOrderIdentities(identities: Array<Identity>): Arra
 	return result;
 }
 
+/**
+ * Return the requested attribute value(s) as an array if present, or an empty array otherwise.
+ *
+ * @param settings - The account settings containing the attributes.
+ * @param key - The key of the attribute in the `attrs` object.
+ * @returns An array containing the attribute value(s) (string | number).
+ */
+export function getAttributeValues<T extends keyof AccountSettings['attrs']>(
+	settings: AccountSettings,
+	key: T
+): AccountSettings['attrs'][T] extends (string | number)[]
+	? (string | number)[]
+	: (string | number)[] {
+	const attributeValue = settings.attrs[key];
+	if (attributeValue !== undefined) {
+		if (Array.isArray(attributeValue)) {
+			return attributeValue as AccountSettings['attrs'][T] extends (string | number)[]
+				? (string | number)[]
+				: never;
+		}
+		return [attributeValue] as (string | number)[];
+	}
+	return [] as (string | number)[];
+}
+
```

Comment:
I'm wondering if it is really needed something like this.
We could have a much simpler function if we move out the reading of the attribute. The util could focus only on returning an array starting from a value that could be both an array or not.
Something like this
```ts
export function asArray<T>(value: T | T[] | undefined): T[] {
	if (value !== undefined) {
		if (Array.isArray(value)) {
			return value;
		}
		return [value];
	}
	return [];
}
```

And then it could be used like this
```ts
result.push(...asArray(settings.attrs.zimbraMailAlias));
result.push(...asArray(settings.attrs.zimbraAllowFromAddress));
```